lib/Controller.js

```diff
@@ -253,14 +259,18 @@ module.exports = (function(){
             return Q.all(stopPromises);
         }
     };
-
+    
     Controller.prototype.getStat = function() {
         return {
             "handledEntries": this.handledEntries,
             "phantomEntries": this.phantomEntries
         }
     };
 
+    Controller.prototype.getAdapter = function() {
```

Comment:
why exposing adapter with the controller?  you can call 
var adapter = require('./Adapters/AdapterFactory.js').getAdapter();
anywhere in code...